docs/en/reference/batch-processing.rst

```diff
@@ -18,14 +18,19 @@ especially what the strategies presented here provide help with.
 
 .. note::
 
-    Having an SQL logger enabled when processing batches can have a serious impact on performance and resource usage.
-    To avoid that you should remove the corresponding middleware.
-    To remove all middlewares, you can use this line:
+    Having an SQL logger enabled when processing batches can have a
+    serious impact on performance and resource usage.
+    To avoid that, you should use a PSR logger implementation that can be
+    disabled at runtime.
+    For example, with Monolog, you can use ``Logger::popHandler()`` as
+    many times as necessary to remove all handlers from the logger.
```

Comment:
You can push a NullHandler instead. The advantage to that approach is that you can pop the NullHandler after your heavy operation and the logger should start logging again.